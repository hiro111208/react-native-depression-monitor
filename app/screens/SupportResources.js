import React from "react";
import {
    ScrollView,
    View,
    StyleSheet,
    Text,
    Linking,
    TouchableOpacity,
} from "react-native";
import colors from "../config/colors";
import * as indexStyles from "../config/indexStyles";

/**
 * Generates support links and contact information for the patient
 */
const SupportResources = ({ navigation }) => {
    return (
        <View style={[styles.container, indexStyles.centering]}>
            <View
                style={[
                    styles.center,
                    indexStyles.shadowEffect,
                    indexStyles.centering,
                    { height: "95%", width: "100%" },
                ]}
            >
                <View
                    style={[
                        {
                            height: "10%",
                        },
                        indexStyles.centering,
                    ]}
                >
                    <Text style={[indexStyles.fontStyle, { fontSize: 22 }]}>
                        Support Resources
                    </Text>
                </View>

                <ScrollView
                    style={{
                        height: "81%",
                    }}
                >
                    <Text
                        style={[
                            indexStyles.centering,
                            {
                                paddingHorizontal: 30,
                                paddingTop: 20,
                                fontSize: 17,
                                fontStyle: "italic",
                                color: "dimgray",
                            },
                        ]}
                        testID={"QUOTE"}
                    >
                        Asking for help doesn't make you weak - it reveals
                        strength even when you don't feel strong.
                    </Text>
                    <View style={{ height: "3%" }}></View>
                    <View
                        style={{
                            height: "6%",
                            width: "60%",
                            backgroundColor: "white",
                            borderTopRightRadius: 20,
                        }}
                    >
                        <Text
                            style={[
                                indexStyles.fontStyle,
                                { fontSize: 18, padding: 5 },
                            ]}
                        >
                            Support Agencies
                        </Text>
                    </View>

                    <View
                        style={[
                            { backgroundColor: colors.lightOutline },
                            indexStyles.centering,
                        ]}
                    >
                        <ScrollView horizontal>
                            <View
                                style={[
                                    { margin: 5 },
                                    styles.boxSmall,
                                    styles.boxMargin,
                                    indexStyles.shadowEffect,
                                ]}
                            >
                                <View
                                    style={[
                                        styles.titleArea,
                                        indexStyles.centering,
                                    ]}
                                >
                                    <Text
                                        style={[
                                            styles.title,
                                            indexStyles.fontStyle,
                                        ]}
                                    >
                                        NHS
                                    </Text>
                                </View>

                                <View
                                    style={[
                                        styles.infoArea,
                                        indexStyles.centering,
                                    ]}
                                >
                                    <Text
                                        style={styles.hyperlinkStyle}
                                        onPress={() => {
                                            Linking.openURL(
                                                "https://www.nhs.uk/service-search/mental-health/find-an-urgent-mental-health-helpline"
                                            );
                                        }}
                                        testID={"NHS"}
                                    >
                                        NHS Helplines
                                    </Text>
                                    <Text />
                                    <Text style={{ fontSize: 14 }}>
                                        Local NHS Urgent Mental Health Helplines
                                    </Text>
                                </View>
                            </View>

                            <View
                                style={[
                                    { margin: 5 },
                                    styles.boxSmall,
                                    styles.boxMargin,
                                    indexStyles.shadowEffect,
                                ]}
                            >
                                <View
                                    style={[
                                        styles.titleArea,
                                        indexStyles.centering,
                                    ]}
                                >
                                    <Text
                                        style={[
                                            styles.title,
                                            indexStyles.fontStyle,
                                        ]}
                                    >
                                        Bipolar UK
                                    </Text>
                                </View>
                                <View
                                    style={[
                                        styles.infoArea,
                                        indexStyles.centering,
                                    ]}
                                >
                                    <Text
                                        style={styles.hyperlinkStyle}
                                        onPress={() => {
                                            Linking.openURL(
                                                "www.bipolaruk.org"
                                            );
                                        }}
                                        testID={"BIPOLAR"}
                                    >
                                        www.bipolaruk.org
                                    </Text>
                                    <Text />
                                    <Text style={{ fontSize: 12 }}>
                                        Supply a range of information leaflets,
                                        books and tapes. Network of self help
                                        groups for people with manic depression,
                                        relatives and friends.
                                    </Text>
                                </View>
                            </View>

                            <View
                                style={[
                                    { margin: 5 },
                                    styles.boxSmall,
                                    styles.boxMargin,
                                    indexStyles.shadowEffect,
                                ]}
                            >
                                <View
                                    style={[
                                        styles.titleArea,
                                        indexStyles.centering,
                                    ]}
                                >
                                    <Text
                                        style={[
                                            styles.title,
                                            indexStyles.fontStyle,
                                        ]}
                                    >
                                        Calmzone
                                    </Text>
                                </View>
                                <View
                                    style={[
                                        styles.infoArea,
                                        indexStyles.centering,
                                    ]}
                                >
                                    <Text
                                        style={styles.hyperlinkStyle}
                                        onPress={() => {
                                            Linking.openURL(
                                                "www.thecalmzone.net"
                                            );
                                        }}
                                        testID={"CALMZONE"}
                                    >
                                        www.thecalmzone.net
                                    </Text>
                                    <Text />
                                    <Text style={{ fontSize: 13 }}>
                                        Campaign Against Living Miserably. Help
                                        and support for young men aged 15-35 on
                                        issues which include depression and
                                        suicide.
                                    </Text>
                                </View>
                            </View>

                            <View
                                style={[
                                    { margin: 5 },
                                    styles.boxSmall,
                                    styles.boxMargin,
                                    indexStyles.shadowEffect,
                                ]}
                            >
                                <View
                                    style={[
                                        styles.titleArea,
                                        indexStyles.centering,
                                    ]}
                                >
                                    <Text
                                        style={[
                                            styles.title,
                                            indexStyles.fontStyle,
                                        ]}
                                    >
                                        Samaritans
                                    </Text>
                                </View>
                                <View
                                    style={[
                                        styles.infoArea,
                                        indexStyles.centering,
                                    ]}
                                >
                                    <Text
                                        style={styles.hyperlinkStyle}
                                        onPress={() => {
                                            Linking.openURL(
                                                "https://www.samaritans.org/how-we-can-help/schools/deal/deal-resources/dealing-feelings/what-is-depression/?gclid=Cj0KCQiAnKeCBhDPARIsAFDTLTIkkHjz1hc7XNw6RtOqswMgB1zpQlKHWeRA5uEqnai06Qw63RGoqQsaAlbJEALw_wcB"
                                            );
                                        }}
                                        testID={"SAMARITANS"}
                                    >
                                        Talk to a Samaritan
                                    </Text>
                                    <Text />
                                    <Text style={{ fontSize: 13 }}>
                                        Learn more about depression and how to
                                        detect it. Contact a Samaritan to have a
                                        chat.
                                    </Text>
                                </View>
                            </View>

                            <View
                                style={[
                                    { margin: 5 },
                                    styles.boxSmall,
                                    styles.boxMargin,
                                    indexStyles.shadowEffect,
                                ]}
                            >
                                <View
                                    style={[
                                        styles.titleArea,
                                        indexStyles.centering,
                                    ]}
                                >
                                    <Text
                                        style={[
                                            styles.title,
                                            indexStyles.fontStyle,
                                        ]}
                                    >
                                        WLM
                                    </Text>
                                </View>
                                <View
                                    style={[
                                        styles.infoArea,
                                        indexStyles.centering,
                                    ]}
                                >
                                    <Text
                                        style={styles.hyperlinkStyle}
                                        onPress={() => {
                                            Linking.openURL(
                                                "https://www.wlm.org.uk/"
                                            );
                                        }}
                                        testID={"WLM"}
                                    >
                                        www.wlm.org.uk
                                    </Text>
                                    <Text />
                                    <Text style={{ fontSize: 13 }}>
                                        WLM Highbury Counselling Centre offers
                                        counselling and psychotherapy service
                                        for adults, a space to speak with a
                                        trained professional.
                                    </Text>
                                </View>
                            </View>
                        </ScrollView>
                    </View>

                    <View style={{ height: "1%" }}></View>

                    <View style={{ alignItems: "center", height: 500 }}>
                        <Text
                            style={[
                                {
                                    fontSize: 16,
                                    padding: 20,
                                    textAlign: "center",
                                },
                            ]}
                        >
                            The research for this project was done by:
                        </Text>
                        <Text
                            style={[
                                indexStyles.fontStyle,
                                { fontSize: 15, textAlign: "center" },
                            ]}
                            testID={"CONTRIBUTORS"}
                        >
                            Jenny Yiend{"\n"}Jong-Sun Lee{"\n"}Sinem Tekes{"\n"}
                            Louise Atkins
                            {"\n"}Andrew Matthews{"\n"}Manouk Vintren{"\n"}
                            Christian Ferragamo
                            {"\n"}Sukhwinder Shergill
                        </Text>
                        <Text />
                        <Text />
                        <Text style={[{ fontSize: 16, padding: 5 }]}>
                            In an academic paper entitled:
                        </Text>
                        <Text
                            style={[
                                indexStyles.fontStyle,
                                {
                                    fontSize: 16,
                                    padding: 10,
                                    textAlign: "center",
                                },
                            ]}
                        >
                            'Modifying Interpretation in a Clinically Depressed
                            Sample Using 'Cognitive Bias Modification-Errors': A
                            Double Blind Randomised Controlled Trial'
                        </Text>
                    </View>
                </ScrollView>

                <View
                    style={[
                        { height: "9%", width: "40%" },
                        indexStyles.centering,
                        indexStyles.shadowEffect,
                    ]}
                >
                    <TouchableOpacity
                        onPress={() => navigation.goBack()}
                        style={[
                            styles.optButton,
                            indexStyles.cover,
                            indexStyles.centering,
                            { borderRadius: 50 },
                        ]}
                        testID={"RETURN_BUTTON"}
                    >
                        <Text style={[indexStyles.fontStyle, { fontSize: 20 }]}>
                            Return
                        </Text>
                    </TouchableOpacity>
                </View>

                <View style={{ height: "1%" }}></View>
            </View>
        </View>
    );
};

const styles = StyleSheet.create({
    infoArea: {
        height: "70%",
        padding: 20,
    },
    bottomBorder: {
        height: "100%",
        width: "40%",
        borderRadius: 50,
        backgroundColor: colors.lightOutline,
    },
    center: {
        backgroundColor: colors.mainPanel,
        borderRadius: 50,
        borderWidth: 5,
        borderColor: colors.lightOutline,
    },
    container: {
        flex: 1,
        display: "flex",
        padding: 25,
        backgroundColor: "#fff",
    },
    optButton: {
        backgroundColor: colors.lightOutline,
    },
    scheduleText: {
        fontSize: 15,
    },
    textStyle: {
        fontSize: 18,
    },
    boxSmall: {
        marginRight: 5,
        width: 225,
        height: 180,
        backgroundColor: "white",
        borderRadius: 40,
        borderWidth: 5,
        borderColor: "#bcf5bc",
    },
    boxLarge: {
        width: 380,
        height: 350,
        backgroundColor: "#ffa351",
    },
    boxMargin: {
        marginRight: 10,
    },
    bigWhite: {
        margin: 15,
        fontSize: 25,
        fontWeight: "bold",
        color: "white",
    },
    hyperlinkStyle: {
        color: "blue",
    },
    title: {
        fontSize: 23,
        fontStyle: "italic",
    },
    titleArea: {
        height: "30%",
        width: "100%",
        backgroundColor: "#bcf5bc",
        borderTopLeftRadius: 30,
        borderTopRightRadius: 30,
    },
});

export default SupportResources;
