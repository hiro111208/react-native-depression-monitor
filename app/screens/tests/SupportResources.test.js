import React from "react";
import { shallow } from "enzyme";
import { render, fireEvent, cleanup } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import SupportResources from "../SupportResources";

describe("Testing SupportResources.js rendering", () => {
    
    it("SupportResources renders correctly", () => {
      const wrapper = shallow(<SupportResources />);
      expect(wrapper.length).toBe(1);
    });
  
    it("SupportResources snapshot test", () => {
      const tree = renderer.create(<SupportResources />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  
    it("Renders quote", () => {
      const wrapper = render(<SupportResources />);
      expect(wrapper.queryByTestId("QUOTE")).toBeTruthy();
    });
  
    it("Renders NHS hyperlink", () => {
      const wrapper = render(<SupportResources />);
      expect(wrapper.queryByTestId("NHS")).toBeTruthy();
    });

    it("Renders BIPOLAR hyperlink", () => {
      const wrapper = render(<SupportResources />);
      expect(wrapper.queryByTestId("BIPOLAR")).toBeTruthy();
    });
  
    it("Renders CALMZONE hyperlink", () => {
      const wrapper = render(<SupportResources />);
      expect(wrapper.queryByTestId("CALMZONE")).toBeTruthy();
    });

    it("Renders SAMARITANS hyperlink", () => {
      const wrapper = render(<SupportResources />);
      expect(wrapper.queryByTestId("SAMARITANS")).toBeTruthy();
    });

    it("Renders WLM hyperlink", () => {
      const wrapper = render(<SupportResources />);
      expect(wrapper.queryByTestId("WLM")).toBeTruthy();
    });

    it("Renders return button", () => {
      const wrapper = render(<SupportResources />);
      expect(wrapper.queryByTestId("RETURN_BUTTON")).toBeTruthy();
    });

    it("Renders contributor list", () => {
      const wrapper = render(<SupportResources />);
      expect(wrapper.queryByTestId("CONTRIBUTORS")).toBeTruthy();
    });

});

describe("Testing SupportResources.js navigation", () => {

  it("Navigate to the previous screen", () => {
    const navigation = { goBack: jest.fn() };
    const wrapper = render(<SupportResources navigation={ navigation } />);
    const button = wrapper.getByTestId("RETURN_BUTTON");
    expect(navigation.goBack).not.toHaveBeenCalled();
    fireEvent(button, "press");
    expect(navigation.goBack).toHaveBeenCalled();
  });

});


afterAll(cleanup); 