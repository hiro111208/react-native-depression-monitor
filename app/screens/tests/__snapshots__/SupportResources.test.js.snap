// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SupportResources.js rendering SupportResources snapshot test 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "display": "flex",
        "flex": 1,
        "padding": 25,
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fed8b1",
          "borderColor": "#ffeed2",
          "borderRadius": 50,
          "borderWidth": 5,
        },
        Object {
          "elevation": 5,
          "marginVertical": 5,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Object {
          "height": "95%",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": "10%",
          },
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "dimgray",
              "fontWeight": "bold",
            },
            Object {
              "fontSize": 22,
            },
          ]
        }
      >
        Support Resources
      </Text>
    </View>
    <RCTScrollView
      style={
        Object {
          "height": "81%",
        }
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "color": "dimgray",
                "fontSize": 17,
                "fontStyle": "italic",
                "paddingHorizontal": 30,
                "paddingTop": 20,
              },
            ]
          }
          testID="QUOTE"
        >
          Asking for help doesn't make you weak - it reveals strength even when you don't feel strong.
        </Text>
        <View
          style={
            Object {
              "height": "3%",
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "white",
              "borderTopRightRadius": 20,
              "height": "6%",
              "width": "60%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "dimgray",
                  "fontWeight": "bold",
                },
                Object {
                  "fontSize": 18,
                  "padding": 5,
                },
              ]
            }
          >
            Support Agencies
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#ffeed2",
              },
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
            ]
          }
        >
          <RCTScrollView
            horizontal={true}
          >
            <View>
              <View
                style={
                  Array [
                    Object {
                      "margin": 5,
                    },
                    Object {
                      "backgroundColor": "white",
                      "borderColor": "#bcf5bc",
                      "borderRadius": 40,
                      "borderWidth": 5,
                      "height": 180,
                      "marginRight": 5,
                      "width": 225,
                    },
                    Object {
                      "marginRight": 10,
                    },
                    Object {
                      "elevation": 5,
                      "marginVertical": 5,
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 2,
                        "width": 0,
                      },
                      "shadowOpacity": 0.25,
                      "shadowRadius": 3.84,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#bcf5bc",
                        "borderTopLeftRadius": 30,
                        "borderTopRightRadius": 30,
                        "height": "30%",
                        "width": "100%",
                      },
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 23,
                          "fontStyle": "italic",
                        },
                        Object {
                          "color": "dimgray",
                          "fontWeight": "bold",
                        },
                      ]
                    }
                  >
                    NHS
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "height": "70%",
                        "padding": 20,
                      },
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                    ]
                  }
                >
                  <Text
                    onPress={[Function]}
                    style={
                      Object {
                        "color": "blue",
                      }
                    }
                    testID="NHS"
                  >
                    NHS Helplines
                  </Text>
                  <Text />
                  <Text
                    style={
                      Object {
                        "fontSize": 14,
                      }
                    }
                  >
                    Local NHS Urgent Mental Health Helplines
                  </Text>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "margin": 5,
                    },
                    Object {
                      "backgroundColor": "white",
                      "borderColor": "#bcf5bc",
                      "borderRadius": 40,
                      "borderWidth": 5,
                      "height": 180,
                      "marginRight": 5,
                      "width": 225,
                    },
                    Object {
                      "marginRight": 10,
                    },
                    Object {
                      "elevation": 5,
                      "marginVertical": 5,
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 2,
                        "width": 0,
                      },
                      "shadowOpacity": 0.25,
                      "shadowRadius": 3.84,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#bcf5bc",
                        "borderTopLeftRadius": 30,
                        "borderTopRightRadius": 30,
                        "height": "30%",
                        "width": "100%",
                      },
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 23,
                          "fontStyle": "italic",
                        },
                        Object {
                          "color": "dimgray",
                          "fontWeight": "bold",
                        },
                      ]
                    }
                  >
                    Bipolar UK
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "height": "70%",
                        "padding": 20,
                      },
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                    ]
                  }
                >
                  <Text
                    onPress={[Function]}
                    style={
                      Object {
                        "color": "blue",
                      }
                    }
                    testID="BIPOLAR"
                  >
                    www.bipolaruk.org
                  </Text>
                  <Text />
                  <Text
                    style={
                      Object {
                        "fontSize": 12,
                      }
                    }
                  >
                    Supply a range of information leaflets, books and tapes. Network of self help groups for people with manic depression, relatives and friends.
                  </Text>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "margin": 5,
                    },
                    Object {
                      "backgroundColor": "white",
                      "borderColor": "#bcf5bc",
                      "borderRadius": 40,
                      "borderWidth": 5,
                      "height": 180,
                      "marginRight": 5,
                      "width": 225,
                    },
                    Object {
                      "marginRight": 10,
                    },
                    Object {
                      "elevation": 5,
                      "marginVertical": 5,
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 2,
                        "width": 0,
                      },
                      "shadowOpacity": 0.25,
                      "shadowRadius": 3.84,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#bcf5bc",
                        "borderTopLeftRadius": 30,
                        "borderTopRightRadius": 30,
                        "height": "30%",
                        "width": "100%",
                      },
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 23,
                          "fontStyle": "italic",
                        },
                        Object {
                          "color": "dimgray",
                          "fontWeight": "bold",
                        },
                      ]
                    }
                  >
                    Calmzone
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "height": "70%",
                        "padding": 20,
                      },
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                    ]
                  }
                >
                  <Text
                    onPress={[Function]}
                    style={
                      Object {
                        "color": "blue",
                      }
                    }
                    testID="CALMZONE"
                  >
                    www.thecalmzone.net
                  </Text>
                  <Text />
                  <Text
                    style={
                      Object {
                        "fontSize": 13,
                      }
                    }
                  >
                    Campaign Against Living Miserably. Help and support for young men aged 15-35 on issues which include depression and suicide.
                  </Text>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "margin": 5,
                    },
                    Object {
                      "backgroundColor": "white",
                      "borderColor": "#bcf5bc",
                      "borderRadius": 40,
                      "borderWidth": 5,
                      "height": 180,
                      "marginRight": 5,
                      "width": 225,
                    },
                    Object {
                      "marginRight": 10,
                    },
                    Object {
                      "elevation": 5,
                      "marginVertical": 5,
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 2,
                        "width": 0,
                      },
                      "shadowOpacity": 0.25,
                      "shadowRadius": 3.84,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#bcf5bc",
                        "borderTopLeftRadius": 30,
                        "borderTopRightRadius": 30,
                        "height": "30%",
                        "width": "100%",
                      },
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 23,
                          "fontStyle": "italic",
                        },
                        Object {
                          "color": "dimgray",
                          "fontWeight": "bold",
                        },
                      ]
                    }
                  >
                    Samaritans
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "height": "70%",
                        "padding": 20,
                      },
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                    ]
                  }
                >
                  <Text
                    onPress={[Function]}
                    style={
                      Object {
                        "color": "blue",
                      }
                    }
                    testID="SAMARITANS"
                  >
                    Talk to a Samaritan
                  </Text>
                  <Text />
                  <Text
                    style={
                      Object {
                        "fontSize": 13,
                      }
                    }
                  >
                    Learn more about depression and how to detect it. Contact a Samaritan to have a chat.
                  </Text>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "margin": 5,
                    },
                    Object {
                      "backgroundColor": "white",
                      "borderColor": "#bcf5bc",
                      "borderRadius": 40,
                      "borderWidth": 5,
                      "height": 180,
                      "marginRight": 5,
                      "width": 225,
                    },
                    Object {
                      "marginRight": 10,
                    },
                    Object {
                      "elevation": 5,
                      "marginVertical": 5,
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 2,
                        "width": 0,
                      },
                      "shadowOpacity": 0.25,
                      "shadowRadius": 3.84,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#bcf5bc",
                        "borderTopLeftRadius": 30,
                        "borderTopRightRadius": 30,
                        "height": "30%",
                        "width": "100%",
                      },
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 23,
                          "fontStyle": "italic",
                        },
                        Object {
                          "color": "dimgray",
                          "fontWeight": "bold",
                        },
                      ]
                    }
                  >
                    WLM
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "height": "70%",
                        "padding": 20,
                      },
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                    ]
                  }
                >
                  <Text
                    onPress={[Function]}
                    style={
                      Object {
                        "color": "blue",
                      }
                    }
                    testID="WLM"
                  >
                    www.wlm.org.uk
                  </Text>
                  <Text />
                  <Text
                    style={
                      Object {
                        "fontSize": 13,
                      }
                    }
                  >
                    WLM Highbury Counselling Centre offers counselling and psychotherapy service for adults, a space to speak with a trained professional.
                  </Text>
                </View>
              </View>
            </View>
          </RCTScrollView>
        </View>
        <View
          style={
            Object {
              "height": "1%",
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 500,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 16,
                  "padding": 20,
                  "textAlign": "center",
                },
              ]
            }
          >
            The research for this project was done by:
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "dimgray",
                  "fontWeight": "bold",
                },
                Object {
                  "fontSize": 15,
                  "textAlign": "center",
                },
              ]
            }
            testID="CONTRIBUTORS"
          >
            Jenny Yiend
            

            Jong-Sun Lee
            

            Sinem Tekes
            

            Louise Atkins
            

            Andrew Matthews
            

            Manouk Vintren
            

            Christian Ferragamo
            

            Sukhwinder Shergill
          </Text>
          <Text />
          <Text />
          <Text
            style={
              Array [
                Object {
                  "fontSize": 16,
                  "padding": 5,
                },
              ]
            }
          >
            In an academic paper entitled:
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "dimgray",
                  "fontWeight": "bold",
                },
                Object {
                  "fontSize": 16,
                  "padding": 10,
                  "textAlign": "center",
                },
              ]
            }
          >
            'Modifying Interpretation in a Clinically Depressed Sample Using 'Cognitive Bias Modification-Errors': A Double Blind Randomised Controlled Trial'
          </Text>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Array [
          Object {
            "height": "9%",
            "width": "40%",
          },
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Object {
            "elevation": 5,
            "marginVertical": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffeed2",
            "borderRadius": 50,
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="RETURN_BUTTON"
      >
        <Text
          style={
            Array [
              Object {
                "color": "dimgray",
                "fontWeight": "bold",
              },
              Object {
                "fontSize": 20,
              },
            ]
          }
        >
          Return
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "height": "1%",
        }
      }
    />
  </View>
</View>
`;
